sharedbuffers/mapped_struct.py

```diff
@@ -438,7 +438,7 @@ def shared_id(obj):
             return obj | LONG_MASK
         elif isinstance(obj, long):
             # Real longs must make room for the flag bits
-            return (obj & 0xFFFFFFFFFFFFFFFF) | ((obj >> 64) << 68) | LONG_MASK
+            return (obj & long(0xFFFFFFFFFFFFFFFF)) | ((obj >> 64) << 68) | LONG_MASK
```

Comment:
BTW... the `long(x)` idiom is only for tests because of this (inefficient). It's ok in tests, we don't care, but actual library code should be efficient.